src/box/memtx_engine.c

```diff
@@ -1125,6 +1125,20 @@ memtx_engine_new(const char *snap_dirname, bool force_recovery,
 			  "increased to 1.001");
 		alloc_factor = 1.001;
 	}
+	/*
+	 * Granularity must be power of 2 and >= 4.
+	 * We can use granularity value == 4 only because we used small
+	 * memory allocator only for struct tuple, which not require aligment.
+	 */
+	unsigned pow2 = 4;
+	while(pow2 < granularity)
+		pow2 <<= 1;
+	unsigned used_granularity = small_align(granularity, pow2);
+	if (used_granularity != granularity) {
+		say_error("Granularity must be multiple of two and >= 4. "
+			  "Changed to %u", used_granularity);
+		granularity = used_granularity;
```

Comment:
To be honest, I'm not sure that we should change specified by user config. I'd better refuse to launch with incorrect value (mb print on exit expected value to simplify user's experience).